### PR TITLE
Rename package to io.opentelemetry.auto

### LINKED ISSUE #7: Rename package to io.opentelemetry.auto
Some things to rename:

* Change package names to `io.opentelemetry.auto.*`
* Change maven artifact `groupId` to `io.opentelemetry.auto`
* Remove or change `dd-` prefix on module names
* Remove or change `DD` prefix on class names

### PR DESCRIPTION
**Pull request by trask**
Closes #7

### CODE REVIEW

File: java-agent/instrumentation/rxjava-1/src/main/java/rx/__OpenTelemetryTracingUtil.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 /**
  * This class must be in the rx package in order to access the package accessible onSubscribe field.
  */
-public class DDTracingUtil {
+public class __OpenTelemetryTracingUtil {
```
Comment (tylerbenson) [inline java-agent/instrumentation/rxjava-1/src/main/java/rx/__OpenTelemetryTracingUtil.java]: Why prefixed?

File: java-agent/agent-tooling/src/main/java/io/opentelemetry/auto/agent/tooling/log/LogContextScopeListener.java [hunk 2]
```diff
@@ -11,8 +11,8 @@
  */
 @Slf4j
 public class LogContextScopeListener {
-  private static final String TRACE_ID_KEY = "dd.trace_id";
-  private static final String SPAN_ID_KEY = "dd.span_id";
+  private static final String TRACE_ID_KEY = "opentelemetry.auto.trace_id";
+  private static final String SPAN_ID_KEY = "opentelemetry.auto.span_id";
```
Comment (tylerbenson) [inline java-agent/agent-tooling/src/main/java/io/opentelemetry/auto/agent/tooling/log/LogContextScopeListener.java]: This key will be used in logging config.  I suggest keeping it short.  `ot.trace_id` should be sufficient.

File: java-agent/instrumentation/rxjava-1/src/main/java/rx/__OpenTelemetryTracingUtil.java (hunk 1 above)
Comment (trask) [reply java-agent/instrumentation/rxjava-1/src/main/java/rx/__OpenTelemetryTracingUtil.java]: I was being excessively(?) worried about conflicting with something rxjava (or someone else) might add into the `rx.` package in the future

File: java-agent/agent-tooling/src/main/java/io/opentelemetry/auto/agent/tooling/log/LogContextScopeListener.java (hunk 2 above)
Comment (trask) [reply java-agent/agent-tooling/src/main/java/io/opentelemetry/auto/agent/tooling/log/LogContextScopeListener.java]: 👍 I will push this change

File: java-agent/instrumentation/rxjava-1/src/main/java/rx/__OpenTelemetryTracingUtil.java (hunk 1 above)
Comment (trask) [reply java-agent/instrumentation/rxjava-1/src/main/java/rx/__OpenTelemetryTracingUtil.java]: It sort of matches the convention of prefixing field names with `__opentelemetry` when we inject fields into other peoples classes.

The name is also weird tho, I will change it to `OpenTelemetryTracingUtil` if you think that's better.

Comment (tylerbenson) [reply java-agent/instrumentation/rxjava-1/src/main/java/rx/__OpenTelemetryTracingUtil.java]: TBH, it doesn't really matter to me.  I can live with it either way.

**Pull request merged**